flow/model/qvalue/avro_converter.go

```diff
@@ -205,103 +212,126 @@ func QValueToAvro(
 	switch v := value.(type) {
 	case types.QValueInvalid:
 		// we will attempt to convert invalid to a string
-		return c.processNullableUnion(v.Val), nil
+		return c.processNullableUnion(v.Val), stringSize(v.Val, nullableSize, calcSize), nil
```

Comment:
I think a good goal here is to do code golf on the extra characters in the switch statement. How about collapsing nullableSize and calcSize into something like `sizeFlag`/`sizeOpt` (`Skip`, `Nullable`, `Plain`) at the cost of little packing on top and unpacking in the utils?